### PR TITLE
Handles closed db connections

### PR DESCRIPTION
**Pull request by dkliban**
When the authentication middleware was added in pulpcore 3.15, it became the first place
in the content app that made an attempt to use the database. As a result, it is a convinient
place to handle InterfaceError and  OperationalError which are raised when the database
connection has been closed. When this occurs, Handler._reset_db_connection() is called to
clean up the database connection and the middleware tries to use the database again.

If the database connection is closed later in the handling of the request by the content app,
the user will still get a 500 error. However, the next request will be handled properly.

This patch also adds a call to Handler._reset_db_connection() inside the heartbeat method.

fixes: #9276
https://pulp.plan.io/issues/9276

Please be sure you have read our documentation on creating PRs:
https://docs.pulpproject.org/contributing/pull-request-walkthrough.html

### CODE REVIEW

File: CHANGES/9598.bugfix [hunk 1]
```diff
@@ -0,0 +1 @@
+Fixed bug where the content app would stop working after a brief loss of connection to the database.
```
Comment (mdellweg) [inline CHANGES/9598.bugfix:1]: Shouldn't it be a backport? Or is the fix on main completely different?

**Comment by mdellweg** [comment]
This PR is made against a stable branch. So i expect the commits to be generated by the cherry-pick-script, which will clearly identify what has been cherry-picked, what was the original issue and what backport issue is fixed.

File: CHANGES/9598.bugfix (hunk 1 above)
Comment (dkliban) [reply CHANGES/9598.bugfix:1]: This is a backport. I did not use the cherry-pick script. In addition to this backport, I had to backport a few other commits.

**Comment by bmbouter** [comment]
I think we do want the clarity the backport script provides, but you can also do these commands manually to ammend your commit to look like what it would produce.

File: CHANGES/9598.bugfix (hunk 1 above)
Comment (bmbouter) [reply CHANGES/9598.bugfix:1]: This updated version is what I expected. Thanks!

**Review by bmbouter — APPROVED** [review]
This looks great. Having done some backports recently, each of these changes looks right to me. Thanks! :1st_place_medal:

**Pull request merged**